### PR TITLE
render: switch to options/config pattern

### PR DESCRIPTION
**Pull request by sttts**
This is the first steps towards a shared render command. In a follow-up we move everything but the cobra command construction into another repo.

### CODE REVIEW

**Comment by sttts** [comment]
/hold

until green smoke test

File: cmd/cluster-kube-controller-manager-operator/render/options/generic.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+// AddFlags adds the generic flags to the flagset.
+func (o *GenericOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.AssetOutputDir, "asset-output-dir", "", "Output path for rendered manifests.")
+	fs.StringVar(&o.AssetInputDir, "asset-input-dir", "", "A path to directory with certificates and secrets.")
+	fs.StringVar(&o.TemplatesDir, "templates-input-dir", "/usr/share/bootkube/manifests", "A path to a directory with manifest templates.")
+	fs.StringSliceVar(&o.ConfigOverrideFiles, "config-override-files", nil, "Additional sparse KubeControllerManagerConfig.kubecontrolplane.config.openshift.io/v1 files for customiziation through the installer, merged into the default config in the given order.")
+	fs.StringVar(&o.ConfigOutputFile, "config-output-file", "", "Output path for the KubeControllerManagerConfig yaml file.")
+
+	// TODO: Remove these once we break the flag dependency loop in installer
+	fs.StringVar(&o.DeprecatedConfigOverrideFile, "config-override-file", "", "")
```
Comment (mfojtik) [inline cmd/cluster-kube-controller-manager-operator/render/options/generic.go]: we can drop this now

File: cmd/cluster-kube-controller-manager-operator/render/options/generic.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+func (o *GenericOptions) configFromDefaultsPlusOverride(templateData interface{}, tlsOverride, bootstrapVersion string, loadAsset LoadAsset) ([]byte, error) {
+	defaultConfig := loadAsset(filepath.Join(bootstrapVersion, "kube-controller-manager", "defaultconfig.yaml"))
+	bootstrapOverrides, err := readFileTemplate(tlsOverride, templateData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load config override file %q: %v", tlsOverride, err)
+	}
+	configs := [][]byte{defaultConfig, bootstrapOverrides}
+
+	// TODO: Remove this when the flag is gone
+	if len(o.DeprecatedConfigOverrideFile) > 0 {
```
Comment (mfojtik) [inline cmd/cluster-kube-controller-manager-operator/render/options/generic.go]: drop

**Comment by mfojtik** [comment]
@sttts two nits, lgtm otherwise

**Comment by sttts** [comment]
Fixed.

/lgtm

**Comment by sttts** [comment]
/hold cancel

smoke test worked.

**Pull request merged**